### ISSUE TITLE
configs/_base.py のセクション分けを目的別に統一

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidetection#675]
## Summary

`configs/_base.py` を含む各 config ファイルのセクション分けを統一し, 「モデル」「データ」「学習」「推論」「デバッグ」「ワークスペース」など目的別に並べ替えて可読性を上げる. キー名や値は変更せず, コメント見出しと並び順のみを整える.

## Background

config 項目が増えたことで `_base.py` の見出しが場当たり的になっており, 関連項目が離れて配置されている. 例:

- `cudnn_benchmark` が「デバイス設定」, `use_fp16` も「デバイス設定」, `enable_tensorboard` も同じセクションに紛れ込んでいる
- `train_score_threshold` と `infer_score_threshold` が同じ「Threshold」ブロックに同居
- `target_classes` が単独セクションで存在するが, 推論パラメータとしてまとめた方が探しやすい
- 「推論ベンチマーク設定」「デバッグ画像保存」「推論時 preprocess 後画像のデバッグ保存」が分散

`pretrained_rtdetr_coco.py` 末尾の「学習設定 (プリトレイン推論では使用しないが, スキーマバリデーション用)」も場当たり対応のままで, セクション体系が決まっていれば自然に収まる.

## Current Implementation

[configs/_base.py](configs/_base.py) のセクション見出し (現状):

```
# クラス設定
# データ設定
# 学習設定
# Early Stopping 設定
# デバイス設定
# Threshold
# 検出対象クラス制限 (推論時のみ. None で全クラス検出)
# 推論ベンチマーク設定
# デバッグ画像保存
# 推論時 preprocess 後画像のデバッグ保存
# Letterbox
# Data Augmentation 設定
# カメラ設定
# ワークスペース設定
```

学習用 / 推論用 / 共通 の境界が不明瞭で, 新しいキーを追加する際にどこに入れるべきか判断に迷う.

## Proposed Changes

`_base.py` を以下の固定セクション構成に並べ替える (案. レビューで調整):

```
# ============================================================
# クラス / データ
# ============================================================
num_classes, class_names, data_root, train_split, val_split

# ============================================================
# 学習
# ============================================================
epochs, lr_scheduler, lr_scheduler_params, early_stopping_*, augmentation, train_score_threshold

# ============================================================
# 推論
# ============================================================
infer_score_threshold, target_classes, letterbox, annotation_path, infer_image_dir

# ============================================================
# デバイス / ランタイム
# ============================================================
device, cudnn_benchmark, use_fp16, enable_tensorboard

# ============================================================
# デバッグ画像保存
# ============================================================
debug_save_count, infer_debug_save_count

# ============================================================
# カメラ (Webcam リアルタイム推論)
# ============================================================
camera_fps, camera_resolution

# ============================================================
# ワークスペース
# ============================================================
work_dir
```

アーキ別 config (`rtdetr_coco.py` / `ssdlite_coco.py` / `ssd300_coco.py` / `pretrained_rtdetr_coco.py`) も同じセクション見出しに統一. セクションが空になる場合はその見出しごと省略する (例: `ssd300_coco.py` に推論セクションは不要).

`pretrained_rtdetr_coco.py` の末尾コメント「学習設定 (プリトレイン推論では使用しないが, スキーマバリデーション用)」は新セクション体系に取り込んで自然な位置に配置する.

## Acceptance Criteria

- `_base.py` が新セクション構成 (上記 7 区分) に並び替わる
- アーキ別 config 4 ファイル (`rtdetr_coco.py`, `ssdlite_coco.py`, `ssd300_coco.py`, `pretrained_rtdetr_coco.py`) も同じ見出し体系に統一される
- 設定キー名 / デフォルト値 / 型に変更が無い (純粋な並び替え + コメント変更のみ)
- 既存の全テスト (`uv run pytest`) が pass
- `uv run pre-commit run --all-files` 全 pass

## Labels

`refactor`

## Branch

`refactor/config-section-layout`
